tests/test_georaster.py

```diff
@@ -33,7 +33,6 @@ def test_init(self):
         # third, rio.Dataset
         ds = rio.open(datasets.get_path("landsat_B4"))
         r3 = gr.Raster(ds)
-        assert r3.filename is None
```

Comment:
Does this mean `r3` will have a filename, or why is this assertion removed?

Reply:
I know read the filename from a `rio.io.DatasetReader`:
```python
self.filename = filename_or_dataset.files[0]
```

Reply:
So should the line instead be changed to:
```python
assert r3.filename is not None
```
maybe? 

Sorry this might not be very important. I just didn't understand why the assertion was removed.

Reply:
I agree with Erik here, it's not clear

Reply:
Added assertion back, it's a detail